modules/plugins/languages/vala.nix

```diff
@@ -0,0 +1,75 @@
+{
+  config,
+  pkgs,
+  lib,
+  ...
+}: let
+  inherit (builtins) attrNames;
+  inherit (lib.options) mkEnableOption mkOption;
+  inherit (lib.modules) mkIf mkMerge;
+  inherit (lib.lists) isList;
+  inherit (lib.types) enum either listOf package str;
+  inherit (lib.nvim.types) mkGrammarOption;
+  inherit (lib.nvim.lua) expToLua;
+
+  cfg = config.vim.languages.vala;
+
+  defaultServer = "vala_ls";
+  servers = {
+    vala_ls = {
+      package = pkgs.writeShellScriptBin "vala-language-server-wrapper" ''
+        export PATH="${lib.makeBinPath [pkgs.uncrustify]}:$PATH"
+        exec ${pkgs.vala-language-server}/bin/vala-language-server "$@"
+      '';
+      internalFormatter = true;
+      lspConfig = ''
+        lspconfig.vala_ls.setup {
+          capabilities = capabilities;
+          on_attach = default_on_attach;
+          cmd = ${
+          if isList cfg.lsp.package
+          then expToLua cfg.lsp.package
+          else ''{"${cfg.lsp.package}/bin/vala-language-server-wrapper"}''
```

Comment:
This won't work if the package isn't wrapped. Either use mkDerivation + wrapProgram or symlinkJoin + makeWrapper in the package above.